test/testUtil.ts

```diff
@@ -152,3 +163,53 @@ export const createFakeConnection = function (): Connection {
     },
   } as any;
 };
+/**
+ * Validate that a CSV file has X records
+ */
+export async function validateCsv(
+  filePath: string,
+  columnDelimiter: ColumnDelimiterKeys,
+  totalQty: number
+): Promise<void> {
+  const csvReadStream = fs.createReadStream(filePath);
+  let recordCount = 0;
+
+  await pipeline(
+    csvReadStream,
+    new csvParse({ columns: true, delimiter: ColumnDelimiter[columnDelimiter] }),
+    new PassThrough({
+      objectMode: true,
+      transform(_chunk, _encoding, callback) {
+        recordCount++;
+        callback(null, null);
+      },
+    }),
+    // dummy writable
+    new Writable({
+      write(_chunk, _encoding, callback) {
+        callback();
+      },
+    })
+  );
+
+  expect(totalQty).to.equal(recordCount);
+}
```

Comment:
test helper to ensure:
1. a valid CSV file was written (it parses it)
2. the total number of records processed by the job were written